components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -223,6 +288,7 @@ public static boolean isSignatureVerified(SignedJWT signedJWT, Certificate x509C
             PublicKey publicKey = x509Certificate.getPublicKey();
             if (publicKey instanceof RSAPublicKey) {
                 verifier = new RSASSAVerifier((RSAPublicKey) publicKey);
+                verifier.getJCAContext().setProvider(BouncyCastleProviderSingleton.getInstance());
```

Comment:
Can we whether this is needed in the runtime? AFAIK this is only needed for unit test. if thats the case lets handle this the same way we have handled in here https://github.com/wso2-extensions/identity-oauth-addons/pull/108#discussion_r1351713879

Reply:
fixed with [404b385](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2156/commits/404b385d6065754da691e95faaac308c2b9d298a)